Library/Homebrew/cask/dsl.rb

```diff
@@ -525,12 +528,26 @@ def livecheckable?
     # NOTE: A warning will be shown when trying to install this cask.
     #
     # @api public
-    def deprecate!(date:, because:, replacement: nil)
+    def deprecate!(date:, because:, replacement: nil, replacement_formula: nil, replacement_cask: nil)
+      if replacement_formula && replacement_cask
+        raise ArgumentError, "replacement_formula and replacement_cask specified!"
+      end
+
+      # TODO: deprecate in >= 4.5.0
+      # if replacement
+      #   odeprecated(
+      #     "deprecate!(:replacement)",
+      #     "deprecate!(:replacement_formula) or deprecate!(:replacement_cask)",
+      #     disable_on: Time.new(2025, 10, 15),
+      #   )
+      # end
+
       @deprecation_date = Date.parse(date)
       return if @deprecation_date > Date.today
 
       @deprecation_reason = because
-      @deprecation_replacement = replacement
+      @deprecation_replacement_formula = replacement_formula
+      @deprecation_replacement_cask = replacement_cask || replacement
```

Comment:
```suggestion
      @deprecation_replacement_formula = replacement_formula.presence || replacement
      @deprecation_replacement_cask = replacement_cask.presence || replacement
```
I think? and same elsewhere